components/tidb_query/src/codec/mysql/decimal.rs

```diff
@@ -2100,14 +2095,11 @@ pub trait DecimalEncoder: NumberEncoder {
     }
 
     fn write_decimal_to_chunk(&mut self, v: &Decimal) -> Result<()> {
-        self.write_u8(v.int_cnt)?;
-        self.write_u8(v.frac_cnt)?;
-        self.write_u8(v.result_frac_cnt)?;
-        self.write_u8(v.negative as u8)?;
-        let len = word_cnt!(v.int_cnt) + word_cnt!(v.frac_cnt);
-        for id in 0..len as usize {
-            self.write_i32_le(v.word_buf[id] as i32)?;
-        }
+        let data = unsafe {
+            let p = v as *const Decimal as *const u8;
+            std::slice::from_raw_parts(p, DECIMAL_STRUCT_SIZE)
```

Comment:
Recommanded to add a [static assert](https://github.com/nvzqz/static-assertions-rs) for `DECIMAL_STRUCT_SIZE` (assert it equals to `size_of(Decimal)`)